Lib/test/test_asyncio/test_futures2.py

```diff
@@ -1,13 +1,68 @@
 # IsolatedAsyncioTestCase based tests
 import asyncio
+import inspect
 import unittest
+import traceback
 
 
 def tearDownModule():
     asyncio.set_event_loop_policy(None)
 
 
 class FutureTests(unittest.IsolatedAsyncioTestCase):
+    maxDiff = None
+
+    async def test_future_traceback(self):
+        # Note: the test is sensitive to line numbers.
+        #
+        # To make the test more flexible
+        # and ready for future file modifications,
+        # RAISE_MARKER and AWAIT_MARKER_<N> are
+        # used for finding analyzed line numbers.
+
+        markers = {}
+
+        def init_markers():
+            source_lines, offset = inspect.getsourcelines(
+                FutureTests.test_future_traceback,
+            )
+            mark = "# pos: "
+            for pos, line in enumerate(source_lines):
+                found = line.find(mark)
+                if found != -1:
+                    marker = line[found + len(mark):].strip()
+                    if marker == '"':
+                        # skip 'mark' variable definition from above
+                        continue
+                    markers[marker] = offset + pos
+
+        init_markers()
+
+        def analyze_traceback(tb, await_marker):
```

Comment:
The traceback tests do this by capturing the line number in a variable and using that. See

https://github.com/python/cpython/blob/e63894b3eed8ad2dd7690695f7f07bfbff59c05a/Lib/test/test_traceback.py#L386

and

https://github.com/python/cpython/blob/e63894b3eed8ad2dd7690695f7f07bfbff59c05a/Lib/test/test_traceback.py#L395

Reply:
It is much simplified now.